testing/file_test/file_test_base.cpp

```diff
@@ -45,6 +65,38 @@ ABSL_FLAG(bool, dump_output, false,
 
 namespace Carbon::Testing {
 
+// Information for a test case.
+struct FileTestInfo {
+  // The name.
+  std::string test_name;
+
+  // A factory function for creating the test object.
+  std::function<auto()->FileTestBase*> factory_fn;
+
+  // gtest's information about the test.
+  ::testing::TestInfo* registered_test;
+
+  // The test result, set after running.
+  std::optional<ErrorOr<TestFile>> test_result;
+
+  // Whether autoupdate would change (or when autoupdating, already changed)
+  // test expectations.
+  bool autoupdate_differs = false;
```

Comment:
Would it be correct to say that this is true if the test output differs from the expected output? If so, it might be a bit clearer to call this something like `output_differs`, and document document it accordingly.